src/Components/Users/UserAdd.tsx

```diff
@@ -245,7 +264,17 @@ export const UserAdd = (props: UserProps) => {
       setIsStateLoading(true);
       const statesRes = await dispatchAction(getStates());
       if (!status.aborted && statesRes.data.results) {
-        setStates([...initialStates, ...statesRes.data.results]);
+        if (userIndex <= USER_TYPES.indexOf("StateAdmin")) {
+          setStates([
+            ...initialStates,
+            {
+              id: currentUser.data.state,
+              name: currentUser.data.state_object.name,
+            },
+          ]);
+        } else {
+          setStates([...initialStates, ...statesRes.data.results]);
+        }
```

Comment:
@cp-Coder @gigincg wouldn't the condition always be true and hence the else block never is used.

`userIndex` is an index within `USER_TYPES`, and StateAdmin is the last element in `USER_TYPES` and hence it has the highest index. so...